lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -298,6 +297,31 @@ public void onAuthUpdated() {
 		}
 	}
 
+	/**
+	 * Called when where was an error during authentication attempt
+	 *
+	 * @param errorInfo Error associated with unsuccessful authentication
+	 */
+	public void onAuthError(ErrorInfo errorInfo) {
+		Log.i(TAG, String.format("onAuthError: (%d) %s", errorInfo.code, errorInfo.message));
+		switch (state.state) {
+			case connecting:
+				ITransport transport = this.transport;
+				if (transport != null)
+					/* onTransportUnavailable will send state change event and set transport to null */
+					onTransportUnavailable(transport, null, errorInfo);
+				break;
+
+			case connected:
+				/* stay connected but notify of authentication error */
+				setState(new StateIndication(ConnectionState.connected, errorInfo));
```

Comment:
Well a connection can also emit an error event, see http://docs.ably.io/client-lib-development-guide/features/#RTN4g.  If the state has not changed, it would be more appropriate to emit that as state changes should only be emitted when there is indeed a state change no?